src/rgw/rgw_asio_frontend.cc

```diff
@@ -611,6 +618,52 @@ int AsioFrontend::get_config_key_val(string name,
   return 0;
 }
 
+bool AsioFrontend::check_secure_mon_conn() const
+{
+  AuthRegistry reg(ctx());
+
+  reg.refresh_config();
+
+  std::vector<uint32_t> methods;
+  std::vector<uint32_t> modes;
+
+  reg.get_supported_methods(CEPH_ENTITY_TYPE_MON, &methods, &modes);
+  ldout(ctx(), 20) << __func__ << "(): auth registy supported: methods=" << methods << " modes=" << modes << dendl;
+
+  for (auto method : methods) {
+    if (!reg.is_secure_method(method)) {
+      ldout(ctx(), 20) << __func__ << "(): method " << method << " is insecure" << dendl;
+      return false;
+    }
+  }
+
+  for (auto mode : modes) {
+    if (!reg.is_secure_mode(mode)) {
+      ldout(ctx(), 20) << __func__ << "(): mode " << mode << " is insecure" << dendl;
+      return false;
+    }
+  }
+
+  return true;
+}
+
+
+void AsioFrontend::warn_if_insecure()
+{
+  if (!maybe_insecure_mon_conn ||
+      warned_insecure) {
+    return;
+  }
+
+  string s = "rgw is configured to optionally allow insecure connections to the monitors (auth_supported, ms_mon_client_mode), ssl certificates stored at the monitor configuration could leak";
+
+  env.store->getRados()->clog_warn(s);
```

Comment:
since we have a service abstraction now, can we move all of this rados-specific stuff into RGWSI_ConfigKey_RADOS?

it looks like the only thing you can't do at that layer is `/* only warn once */`, but i don't think that's really necessary

Reply:
re: warn-once, i guess you could just store that bool with the service

Reply:
@cbodley the config key svc is not really an appropriate place for it, and I was avoiding creating a new svc

Reply:
why isn't it appropriate? this is what we created it for

Reply:
@cbodley it was creating to store/retrieve config keys, not to distribute cluster-wide messages

Reply:
we want a service that provides secure access to secrets. the cluster warning is part of our security design for that, so it's an implementation detail of `RGWSI_ConfigKey_RADOS`. same with the use of `AuthRegistry::get_supported_methods()`

Reply:
@cbodley ok, changed accordingly